src/test/java/org/opensearch/flowframework/workflow/CreateIndexStepTests.java

```diff
@@ -89,4 +120,67 @@ public void testCreateIndexStepFailure() throws ExecutionException, InterruptedE
         assertTrue(ex.getCause() instanceof Exception);
         assertEquals("Failed to create an index", ex.getCause().getMessage());
     }
+
+    public void testInitIndexIfAbsent_IndexNotPresent() {
+        when(metadata.hasIndex(FlowFrameworkIndex.GLOBAL_CONTEXT.getIndexName())).thenReturn(false);
+
+        ActionListener<Boolean> listener = mock(ActionListener.class);
```

Comment:
```suggestion
                @SuppressWarnings("unchecked")
                ActionListener<Boolean> listener = mock(ActionListener.class);
```
And the other mocks in this class of ActionListener and Map.